### ISSUE TITLE
Original Java exception backtrace lost

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#7759]
In some of our Java code which  is wrapped by a JRuby extension, in some places we use code to *convert* a Java exception to a Ruby one with code like this:

```java
        @JRubyMethod(name = "open")
        public IRubyObject ruby_open(ThreadContext context)
        {
            try {
                this.queue.open();
            } catch (IOException e) {
                throw context.runtime.newIOErrorFromException(e);
            }

            return context.nil;
        }
```

The problem is that the original `IOException` in this case, is rethrown using `throw context.runtime.newIOErrorFromException(e);` and the higher-up backtrace logging looses the original exception backtrace and logs the new one. 

For example below, a forced exception was added in the Java `Queue.open()` method and this is the backtrace we see:

```
[2017-07-20T13:47:27,216][ERROR][logstash.pipeline        ] Logstash failed to create queue {:pipeline_id=>"main", "exception"=>"", "backtrace"=>["org/logstash/ackedqueue/ext/JrubyAckedQueueExtLibrary.java:133:in `open'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:41:in `with_queue'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:31:in `create_file_based'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/queue_factory.rb:29:in `create'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/pipeline.rb:200:in `initialize'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/pipeline_action/create.rb:35:in `execute'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:342:in `block in converge_state'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:141:in `with_pipelines'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:339:in `block in converge_state'", "org/jruby/RubyArray.java:1734:in `each'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:326:in `converge_state'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:166:in `block in converge_state_and_update'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:141:in `with_pipelines'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:164:in `converge_state_and_update'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/agent.rb:90:in `execute'", "/Users/colin/dev/src/elasticsearch/logstash/logstash-core/lib/logstash/runner.rb:346:in `block in execute'", "/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/2.3.0/gems/stud-0.0.22/lib/stud/task.rb:24:in `block in initialize'"]}
[2017-07-20T13:47:27,218][ERROR][logstash.agent           ] Failed to execute action {:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"IOError", :message=>""}
```

We should find a way to report the original exception information which will be a lot more informative when troubleshooting problems.

**Comment by colinsurprenant** [comment]
PR in #8245

**Comment by colinsurprenant** [comment]
closed by #8245 which was merged into master, 6.x and 6.0.
